src/state.c

```diff
@@ -284,6 +287,37 @@ handle_state_mounts_section(GNode* node, struct handler_data* data) {
 }
 
 /*!
+ * handler for namespaces section
+ *
+ * \param node \c GNode.
+ * \param data \ref handler_data.
+ */
+static void
+handle_state_namespaces_section(GNode* node, struct handler_data* data) {
+	struct oci_cfg_namespace* n;
+
+	if (! (node && node->data)) {
+		return;
+	}
+	if (! (node->children && node->children->data)) {
+		g_critical("%s missing value", (char*)node->data);
+		return;
+	}
+
+	if (! g_strcmp0(node->data, "type")) {
+		n = g_new0 (struct oci_cfg_namespace, 1);
+		n->type = cc_oci_str_to_ns((char*)node->children->data);
+		data->state->namespaces = g_slist_append(data->state->namespaces, n);
```

Comment:
according to documentation https://developer.gnome.org/glib/2.26/glib-Singly-Linked-Lists.html#g-slist-append

```
g_slist_append ()
Returns : the new start of the GSList
```